lib/mocha.js

```diff
@@ -244,14 +244,22 @@ Mocha.prototype._growl = function (runner, reporter) {
 
   runner.on('end', function () {
     var stats = reporter.stats;
+    var errMsg = 'The prerequisites for Growl support are missing. Please see https://github.com/mochajs/mocha/wiki/Growl-Notifications for details';
+    var msg = '';
+
     if (stats.failures) {
-      var msg = stats.failures + ' of ' + runner.total + ' tests failed';
-      notify(msg, { name: 'mocha', title: 'Failed', image: image('error') });
+      msg = stats.failures + ' of ' + runner.total + ' tests failed';
+      notify(msg, { name: 'mocha', title: 'Failed', image: image('error') }, function (err) {
+        if (err && err.code === 'ENOENT') {
+          console.warn(errMsg);
```

Comment:
If this is *not* `ENOENT`, we should not eat the error; please throw it.

Reply:
Agreed.

Reply:
The error is already being thrown elsewhere, the callback parameter in `growl()` is used to add additional error reporting.  There is no need to throw an error inside the callback (note, the default callback is a noop function).

Reply:
@plroebuck The `notify` function (`growl`) takes in 3 parameters, the third of which being a callback that is fired when growl finishes. That callback is being fired from an event on the child process created by `growl`, essentially in a separate thread. Adding logic to rethrow that error results in no change to the output. Currently, a callback is not being passed in at all, causing growl to assign a noop function to the callback, meaning the error that growl returns in the callback is already being swallowed.

Reply:
Sench has a point, though Growl errors aren't swallowed as they do show up in Mocha's output.

However, Growl support is tangential to Mocha's purpose. Rethrowing Growl errors doesn't buy us anything. As such, I think we should do something like:

```js
Mocha.prototype._growl = function (runner, reporter) {
  var notify = require('growl');
  var sendNotification = function () {
    var image = function (name) {
      return path.join(__dirname, '../images', name + '.png');
    };
    var onNotify = function (err) {
      if (err) {
        if (err.code === 'ENOENT') {
          process.stderr.write('Growl notification support available but not installed -- disabling!\n');
          runner.removeListener('end', sendNotification);
        } else {
          process.stderr.write('growl: ' +  err.message + '\n');
        }
      }
    };
    var stats = reporter.stats;
    var msg;

    if (stats.failures) {
      msg = stats.failures + ' of ' + runner.total + ' tests failed';
      notify(msg, { name: 'mocha', title: 'Failed', image: image('error') }, onNotify);
    } else {
      msg = stats.passes + ' tests passed in ' + stats.duration + 'ms';
      notify(msg, { name: 'mocha', title: 'Passed', image: image('ok') }, onNotify);
    }
  };
  runner.on('end', sendNotification);    // should this be `.once`?
};
```

Of course this leads me to ask how many `end` events we expect to process anyway.
If that answer is one, removing the listener may be too little, too late.

It's unfortunate that the `growl` package doesn't allow for solving this problem without attempting notification. [I asked](https://github.com/tj/node-growl/issues/74)... without response. Much better solved in `Mocha#growl` than `Mocha#_growl`.